core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -556,18 +556,30 @@ public RequestLogBuilder logBuilder() {
 
     @Override
     public String toString() {
+        final String clientRequestIdStr = toStringClientRequestId();
         if (strVal != null) {
-            return strVal;
+            return clientRequestIdStr + strVal;
         }
-        return toStringSlow();
+        return clientRequestIdStr + toStringSlow();
+    }
```

Comment:
This will create a new `clientReqIdStr` every time. How about adding some state variable to minimize the regeneration of strings? e.g. You could create a small flag that keeps whether 1) `channel` is available and 2) `log.parent` is available, so `strVal` is regenerated when the flag changes.

Reply:
Oh, that sounds nice. Let me address it!